Small menu changes/Overall improvements

Deadlineem opened the pull request:
Added a few things and changed a few things, nothing major.

One thing about this, you can click/drag the menu anywhere on your screen now, however, the click/drag functions on the entire menu window and not just the title bar, if someone knows how to fix that?

Game version is manually added, dont see much of a need to change it from that but it could be dynamic so its out of our hair?

Rounded off the menu, added a show/hide toggle to it and title bar, resized slightly.

![Screenshot_16](https://github.com/YimMenu/HorseMenu/assets/24372625/c7d6c419-79ff-4880-b743-f33d378cf7f1)
![Screenshot_17](https://github.com/YimMenu/HorseMenu/assets/24372625/86293bab-81b1-4173-8ce3-4d30b78f513a)


EDIT: Updated UIManager to retrieve the games build automatically, see screenshot.

![Screenshot_18](https://github.com/YimMenu/HorseMenu/assets/24372625/32dd03e2-d3d4-45aa-ac11-343588a22eda)

DayibBaba commented on src/game/frontend/Menu.cpp:
```diff
@@ -1,5 +1,6 @@
 #include "Menu.hpp"
 
+#include "game/rdr/Natives.hpp"
```
Don't see why you would include this and not use it.

Deadlineem replied:
I was messing with PAD::DISABLE_CONTROL_ACTION to attempt to stop the game from being controlled while the menu is opened, thats why its there.

Deadlineem replied:
> I would remove it, as it is an unnecessary import

Could, but im still working on that bit on my end, mainly because when the menu is opened, clicking on options makes the player shoot their weapons, also when scrolling through the menu options if there is a scroll to it, it will scroll through the game options (on the main screen)

So disabling that stuff is what im after

Deadlineem replied:
Ik hes pretty busy, but maybe @Yimura could help achieve this.  I tried putting the native in for disabling control actions in-game when the menu is opened but it entirely disables the mouse in game, regardless if menu is opened or not. 
ik im doing something wrong so before i pushed the commit i removed the disable control portion, but left the native include.

DayibBaba reviewed (approved):
Ill consider the game version as a placeholder for now.

Rxann commented on src/core/frontend/manager/UIManager.cpp:
```diff
@@ -31,7 +31,7 @@ namespace YimMenu
 		ImGui::EndChild();
 		
-		ImGui::Text("HorseMenu");
+		ImGui::Text("Game Ver. 1.32"); // Game version
```
You could do something like 
`HorseMenu - Game Ver. 1.32`

Deadlineem replied:
True, but it seems unnecessary to add horsemenu at the bottom text with the new change.

Rxann replied:
yeah, either or it. It is a small detail so it doesn't matter much. Are there any natives to get the game version? those could be used if there are

Deadlineem replied:
Not sure but i will check tonight and see.  Had some PC issues from a windows update when i woke up today so reinstalled windows before i left for work.

Deadlineem commented:
@DayibBaba @Rxann 

DEBUG::GET_GAME_VERSION_NAME() returns 1491.50_dev_live_tu when its used, which is the correct title update being retrieved however, its suuuuper long.

Not sure if theres a way in C++ to remove the _dev_live_tu portion so the output is 1491.50

Rxann commented:
> @DayibBaba @Rxann
> 
> DEBUG::GET_GAME_VERSION_NAME() returns 1491.50_dev_live_tu when its used, which is the correct title update being retrieved however, its suuuuper long.
> 
> Not sure if theres a way in C++ to remove the _dev_live_tu portion so the output is 1491.50

Yes there is. Use erase. It takes a position to remove, so just count up to the _ and then put that in. The position probably starts at 0 due to this being c++

Deadlineem commented:
Updated UIManager to dynamically retrieve the games build information and display it.  Thanks @Rxann for the guidance with that.

Deadlineem commented:
Removed native include from Menu.cpp as @DayibBaba suggested, it was unneeded

Added native include to GUI.cpp for using ```PAD::DISABLE_ALL_CONTROL_ACTIONS(0);``` when the menu is opened, no longer controls the player actions like shooting when navigating through the menu.

DayibBaba commented:
> Removed native include from Menu.cpp as @DayibBaba suggested, it was unneeded
> 
> Added native include to GUI.cpp for using `PAD::DISABLE_ALL_CONTROL_ACTIONS(0);` when the menu is opened, no longer controls the player actions like shooting when navigating through the menu.

Blocking all controls is undesirable. I'd rather be able to move. Youll have to see what actions are unneeded when menu is opened, and block them manually. My initial control blocking does most of it already.

Deadlineem commented:
> > Removed native include from Menu.cpp as @DayibBaba suggested, it was unneeded
> > Added native include to GUI.cpp for using `PAD::DISABLE_ALL_CONTROL_ACTIONS(0);` when the menu is opened, no longer controls the player actions like shooting when navigating through the menu.
> 
> Blocking all controls is undesirable. I'd rather be able to move. Youll have to see what actions are unneeded when menu is opened, and block them manually. My initial control blocking does most of it already.

Im still able to move on my end with it, for both keyboard and controller, just disables shooting while menu is opened and not in a wagon (idky the wagon).  Can you test this on your end?
